### ISSUE TITLE
Is it allowed to host a customized version of the Hanabi-live myself?

### ISSUE THREAD

**Issue by hengyuan-hu** [Zamiell/hanabi-live#2416]
Hi, thanks a lot for the amazing code. I would like to host a customized version of this on AWS. The customization is mainly some hack to make it easier to implement our artificial bots and a few extra commands to allow running tests between the bots. We have been using it internally and want to host a public version to allow others to play with the bots.

The code is open-sourced so I guess anyone can use it and setup a new server somewhere. But I still want to check with you to see if it is not allowed or strongly discouraged.

**Comment by Zamiell** [comment]
why not submit a pull request here so that everyone can benefit from the improvements?
and then, you can direct people to play with your bots on the official version

**Comment by hengyuan-hu** [comment]
Haha, I would not really call them improvements. For example, the most important/hacky thing I did was to send the card info of player's own cards to the player as part of the json because that's necessary to construct some game state object of another library commonly used for training AI bots for Hanabi. This is fine for us as the info is not rendered, but I am not sure whether it is commonly acceptable for the official version. Many stuff we did on the implementation of the bot based on a server checkpoint of mid 2020. I tried to switch to a newer main branch a few months ago but struggled to make it work. I am not very good at this stack of tech :(

We have some tools that we can only run from the server side, such as automatically running 500 games between some bots to test if the implementation is correct by looking at average scores. I am not sure how that could done without access to the server myself.

We also want to access all the games played by different bots to run some analysis scripts. For the comparison purpose, we want to use fixed sequence of seeds for all bots to reduce variance, not sure that's a good idea for normal game play.

**Comment by Zamiell** [comment]
> This is fine for us as the info is not rendered, but I am not sure whether it is commonly acceptable for the official version.

indeed, that is not desired since it increases the vector for players to be cheating

> We also want to access all the games played by different bots to run some analysis scripts. For the comparison purpose, we want to use fixed sequence of seeds for all bots to reduce variance, not sure that's a good idea for normal game play.

ok, so it does sound like maybe the best course of action is to have a separate website dedicated for bots.
is the intent to only play 2-player games with 1 bot and 1 human?
perhaps you could modify the branding of the website to clearly indicate this, and add code to prevent all-human games from being started.

**Comment by hengyuan-hu** [comment]
> perhaps you could modify the branding of the website to clearly indicate this
Yeah I would surely change the name, like Hanabot or something. I can also add text in the login page to encourage the use of the official site for non-bot related games.

It can be 2-3 player games and it can be 2 human 1 bot, but we only use the most basic 5 color 5 rank version in the current multi-agent research community. I can disable as many variants as possible to make it less interesting for the general public.

Currently I identify bots only through names (i.e. whether they contain the word "bot"). So I don't really know how to prevent all-human games if some of them call themselves bot. I can set a restriction that at least one player's name must contain "bot".

**Comment by timotree3** [comment]
> The code is open-sourced so I guess anyone can use it and setup a new server somewhere. But I still want to check with you to see if it is not allowed or strongly discouraged.

To be clear, as long as you make your code changes open source, you're allowed to use this code for whatever you want. Zamiel may have requests about how you brand it, but those are all optional. There is nothing wrong or discouraged about setting up your own server.

**Comment by Zamiell** [comment]
> Currently I identify bots only through names (i.e. whether they contain the word "bot"). So I don't really know how to prevent all-human games if some of them call themselves bot.

could you add a DB field to the "users" table that identifies a bot as a bot?
it seems like that would be pretty easy to do

**Comment by hengyuan-hu** [comment]
Thanks for the suggestion! I will look into that.
